meetings/2022-05-16.md

```diff
@@ -0,0 +1,95 @@
+* AC: Note that the WebID and the Client ID are being conflated in this case. How can we support this? One idea would be: it you have an existing Client ID document, and augmenting (potentially using a secondary JSON-LD context). You could use this context to reference key material. The server could generate a nonce as a challenge for the client to demonstrate possession of a private portion of the public key material available in their client ID/WebID document.
+* EP: I think having apps acting on their own behalf could be an antipattern. There exists a distinction between clients and social agents which is described in the interoperability panel. An example is provided in the alternative flow document.
+* AC: What you describe is a very typical use case (app acting on behalf of a social agent) can be implemented using offline_access scope and refresh tokens. App acting on its own behalf is a different use case which isn't covered by Solid-OIDC currently. 
+* NC: When you receive the refresh token you set timeout to keep access token refreshed as long as the session is active.
+* EP: So refresh tokens itself don't have expiration?
+* NC: Token response includes access token and refresh token which are opaque to the client and information about expiration for the access token.
+* NC: When you restore persisted session, the refresh token is used to get new access token.
+* EP: I see there is assumption that refresh tokens are long lasting, weeks not hours or days.
+* AC: Using refresh tokens also often relies on the assumtion that some persistent storage layer is available.
+* EP: There are proposed solutions in `alternative-flows.md`. In order to implement them and support the discussed use case, should the Solid-OIDC spec be extended, or should new specs altogether be proposed?
+* AC: A difficulty we'll be facing is that device code flow/client credentials flow are OAuth2.0 contructs, while the ID token is an OIDC construct, defined at a different layer. We'll want to keep these concerns separate. Writing our own services makes that possible, but it will limit adoption of third-party software (e.g. CSS has faced issues implementing the client credential flow). Defining these will need to happen at the right level of abstraction.
+* EP: +1. The first proposed solition (oauth token exchange) involves exchanging an oauth access token for an id token, so the distinction exists there. Same for using UMA, where the AS takes the ID token as a claim and returns any token in response. The AS would need to accept additional claims in order to support the use case we've described. Given this, is it interesting at all to keep using the ID token as a claim at all?
+* AC: Both 1 & 2nd solution seems promising. It would require to define JWT format. If this format is used in context of token exchange we explain how it gets used there. If it gets pushed to UMA server we also would define it.
```

Comment:
```suggestion
* AC: Both first & second solutions seem promising. It would require that we define JWT format. If this format is used in context of token exchange, we explain how it gets used there. If it gets pushed to UMA server, we also would define it.
```